s3/pkg/datastore/yig/meta/db/tidb/driver.go

```diff
@@ -0,0 +1,36 @@
+package tidb
+
+import (
+	"database/sql"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/opensds/multi-cloud/s3/pkg/datastore/yig/config"
+	"github.com/opensds/multi-cloud/s3/pkg/datastore/yig/meta/db/driver"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	TIDB_DRIVER_ID = "tidb"
+)
+
+type TidbDriver struct {
+}
+
+func (td *TidbDriver) OpenDB(dbCfg config.DatabaseConfig) (driver.DB, error) {
```

Comment:
please ensure this method is idempotent

Reply:
@rhsakarpos  thanks for your review, and what do you mean the idempotent in this function? and original object of this function is to create a new db handle for caller usage. Each call to openDB will create a new db handle.

Reply:
idempotent will mean, no matter how many times the function is called, create only one DB handle...something like
```
synchronized()
{
    if(dbHandle == null)
    {
        createDBHandle()
    }
}
```
Depends on the need...do we need only one DB handle across the runtime of the application or many

Reply:
@rhsakarpos thanks for your good explanation, and for one yig backend instance, there is only one db handle, and this is controlled by the upper logic, so, in our scenario, yig backend will control to create only one db handle for usage. and for above OpenDB function, it will always return a new db handle when it is called.